contracts/src/AgentExecutor.sol

```diff
@@ -36,4 +48,19 @@ contract AgentExecutor {
     function _transferToken(address token, address recipient, uint128 amount) internal {
         IERC20(token).safeTransfer(recipient, amount);
     }
+
+    /// @dev Register native asset from polkadto as ERC20 `token`.
+    function _registerToken(bytes32 agentID, bytes32 tokenID, string memory name, string memory symbol, uint8 decimals)
+        internal
+    {
+        IERC20 token = new ERC20(name, symbol, decimals);
+        Gateway(msg.sender).registerTokenByID(tokenID, address(token), agentID);
```

Comment:
Thought experiment:

Instead of calling a function on the Gateway, make `_registerToken` and `execute` return an ABI-encoded `calldata`. Then in `Gateway.agentExecute`, execute that `calldata` using `call`.

Reply:
https://github.com/Snowfork/snowbridge/pull/1155/commits/780c7bd920796d22169e34406ff48ea8b4b041cf

Reply:
Reverted.